lib/src/package_analysis.dart

```diff
@@ -0,0 +1,60 @@
+import 'dart:io';
+
+import 'package:analyzer/dart/analysis/context_builder.dart';
+import 'package:analyzer/dart/analysis/context_locator.dart';
+import 'package:analyzer/dart/analysis/results.dart';
+import 'package:analyzer/dart/element/element.dart';
+import 'package:args/args.dart';
+import 'package:path/path.dart' as path;
+
+Future<void> main(List<String> arguments) async {
+  // parse command-line arguments
```

Comment:
The analyser will have a number of distinct functions (creating a JSON summary, comparing two JSON summaries, comparing two package versions directly), so it probably makes sense to introduce subcommands for each of these functions